File: nifi-starter-processors/src/test/java/org/example/processors/starter/MyProcessorTest.java
package org.example.processors.starter;

import org.apache.nifi.util.TestRunner;
import org.apache.nifi.util.TestRunners;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class MyProcessorTest {

  private TestRunner testRunner;

  @BeforeEach
  public void init() {
    testRunner = TestRunners.newTestRunner(MyProcessor.class);
  }

  @Test
  public void testProcessor() {}
}

File: nifi-starter-processors/src/test/kotlin/org/example/processors/starter/MyProcessorTest.kt
package org.example.processors.starter

import org.apache.nifi.util.TestRunner
import org.apache.nifi.util.TestRunners
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MyProcessorTest {

  private var testRunner: TestRunner? = null

  @BeforeEach
  fun init() {
    testRunner = TestRunners.newTestRunner(MyProcessor::class.java)
  }

  @Test fun testProcessor() {}
}